src/Symfony/Bundle/TwigBundle/DependencyInjection/Compiler/DefaultOverriddenBundlePathPass.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\TwigBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\Config\Resource\FileExistenceResource;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+/**
+ * Registers the default overriding bundles paths.
+ */
+final class DefaultOverriddenBundlePathPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container)
+    {
+        $twigLoaderFilesystemId = 'twig.loader.native_filesystem';
+
+        if (false === $container->hasDefinition($twigLoaderFilesystemId)) {
+            return;
+        }
+
+        $twigLoaderFilesystemDefinition = $container->getDefinition($twigLoaderFilesystemId);
+        $twigDefaultPath = $container->getParameter('twig.default_path');
+
+        foreach ($container->getParameter('kernel.bundles_metadata') as $name => $bundle) {
+            $defaultOverrideBundlePath = $container->getParameterBag()->resolveValue($twigDefaultPath).'/bundles/'.$name;
+
+            if (file_exists($dir = $container->getParameter('kernel.root_dir').'/Resources/'.$name.'/views')) {
+                @trigger_error(sprintf('Templates directory "%s" is deprecated since Symfony 4.2, use "%s" instead.', $dir, $defaultOverrideBundlePath), E_USER_DEPRECATED);
+
+                $twigLoaderFilesystemDefinition->addMethodCall(
+                    'prependPath',
```

Comment:
I afraid prepending the path is a BC break if your current project has user-configured paths for some of the installed bundles, i.e. before, user-configured paths are at the top of the list (high priority), after they will be under these "override" paths. That mean, someone having 2 versions of the same template (e.g. the process of redesign templates when we're migration from EA1 to EA2) will be affected for this change of priority.

For instance, this is the current paths order for namespaces in one app:
 G1. User-configured path (`app/Resources/views/easyadmin/`) 
 G2. "Override" paths (`templates/bundles/EasyAdminBundle/`)
 G3. The bunde `views` path (`vendor/easycorp/easyadmin-bundle/src/Resources/views/`)

I think you should keep this registration order to avoid breaking BC.

Now, to achieve the proposal, you can separate the groups 2 and 3 into individual compiler passes with different priorities (i.e. G2 = 1 => early and G3 = -1 => late) thus, 3rd-party bundle can register their paths between G2 and G3.

Reply:
In the other hand, you have an automatic solution to this problem by adding the "override" EasyAdmin path again into `twig.paths` config, implementing `PrependExtensionInterface` in your admin-bundle extension. Or manually at project level `config/packages/twig.yaml`:
```
/var/www/project/templates/bundles/EasyAdminBundle/  (added by AdminBundle)
/var/www/project/vendor/kiplin/admin-bundle/src/Resources/views/
/var/www/project/vendor/alterphp/easyadmin-extension-bundle/src/Resources/views/
/var/www/project/templates/bundles/EasyAdminBundle/ (never will be reached, first wins)
/var/www/project/vendor/easycorp/easyadmin-bundle/src/Resources/views/
```

Reply:
Hi @yceruto, thanks for your feedback. How do you define user-configured paths (G1) ? I'm not sure to understand if you are talking about Symfony templates overriding or specific way of EasyAdmin to deal with it (which was not standard in EA1).

Reply:
Nothing about EA1 convention paths, I meant this:
```yaml
# config/packages/twig.yaml
twig:
    paths: # User-configured paths
        templates/bundles/EasyAdminBundle/: EasyAdmin
```

Reply:
Ok, I understand now. I did not know this possibility.

Reply:
I will try to handle all those cases !

Reply:
@yceruto User-configured paths are handled, and the order is preserved (first user defined has priority on last user defined one, for same namespace).